File: BackEnd/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/Struttura.java
package it.unicam.cs.CasottoIdS.models;

import org.springframework.data.annotation.Id;

public class Struttura {

    @Id
    public String idStruttura;
    private String nome;
    private StatoStruttura stato;

    public Struttura(String nome, StatoStruttura stato) {
        this.nome = nome;
        this.stato = stato;
    }

    public String getIdStruttura() {
        return idStruttura;
    }

    public void setIdStruttura(String idStruttura) {
        this.idStruttura = idStruttura;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public StatoStruttura getStato() {
        return stato;
    }

    public void setStato(StatoStruttura stato) {
        this.stato = stato;
    }
}

File: FrontEnd/Casotto/lib/models/Struttura.dart
import 'StatoStruttura.dart';

class Struttura {
  late String idStruttura;
  late String nome;
  late StatoStruttura stato;

  Struttura(String idStruttura, String nome, StatoStruttura stato) {
    this.idStruttura = idStruttura;
    this.nome = nome;
    this.stato = stato;
  }

  String getIdStruttura() {
    return this.idStruttura;
  }

  String getNome() {
    return this.nome;
  }

  void setNome(String newNome) {
    this.nome = newNome;
  }

  StatoStruttura getStatoStruttura() {
    return this.stato;
  }

  void setStatoStruttura(StatoStruttura newStatoStruttura) {
    this.stato = newStatoStruttura;
  }

  String getStatoStrutturaString() {
    switch (this.stato) {
      case StatoStruttura.LIBERA:
        return "LIBERA";
      case StatoStruttura.OCCUPATA:
        return "OCCUPATA";
    }
  }

  @override
  String toString() {
    return "Struttura{" +
        "idStruttura='" +
        idStruttura +
        '\'' +
        ", nome=" +
        nome +
        ", stato" +
        stato.toString() +
        '}';
  }
}
